skimage/io/_plugins/matplotlib_plugin.py

```diff
@@ -141,14 +140,15 @@ def imshow(im, *args, **kwargs):
     ax_im : `matplotlib.pyplot.AxesImage`
         The `AxesImage` object returned by `plt.imshow`.
     """
+    if kwargs.get('cmap', None) == 'viridis':
+        kwargs['cmap'] = viridis
     lo, hi, cmap = _get_display_range(im)
     kwargs.setdefault('interpolation', 'nearest')
     kwargs.setdefault('cmap', cmap)
     kwargs.setdefault('vmin', lo)
     kwargs.setdefault('vmax', hi)
     ax_im = plt.imshow(im, *args, **kwargs)
-    if cmap != _default_colormap:
-        plt.colorbar()
+    plt.colorbar()
```

Comment:
This needs at least some conditioning, or we'll get some weird behavior with RGB images, no?